### PR TITLE
Add Redox Data Models

### PR DESCRIPTION
**Pull request by toddkazakov**

### CODE REVIEW

File: reference/redox.v1.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+openapi: 3.0.0
+x-stoplight:
+  id: gpm2ktdetzht7
+info:
+  title: Redox
+  version: '1.0'
+  description: |-
+    This spec is used for generating models for incomming Redox messages. In the future it may also be used for generating an SDK library for pushing results to Redox.
```
Comment (tjotala) [inline reference/redox.v1.yaml:8]: Nit: `incoming`

File: reference/redox.v1.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      description: |-
+        Currently, this endpoint's only purpose is to reference Redox models so we can use them for code generation. The code generation tool emits models only if they are referenced in the API portion of the spec.
+
+        In the future we may use this for sending data to Redox as well.
+    parameters: []
+components:
+  schemas:
+    NewOrder:
+      $ref: ./redox/order/new.yaml
+  securitySchemes: {}
```
Comment (tjotala) [inline reference/redox.v1.yaml:35]: I guess as long as we're not actually using this to actually send a request to Redox, we don't need a definition of the `securitySchemes`?

File: reference/redox.v1.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    To enable code generation for a specific model you should reference it in the request body `/endpoint` API endpoint.
+servers:
+  - url: 'https://api.redoxengine.com/endpoint'
+paths:
+  /endpoint:
+    post:
+      summary: Send Data
+      tags: []
+      responses: {}
```
Comment (tjotala) [inline reference/redox.v1.yaml:18]: Should we include the definition of a response? https://docs.redoxengine.com/basics/data-exchange-with-redox/responding-to-asynchronous-events/#responses-to-send

File: reference/redox.v1.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      operationId: SendData
+      requestBody:
+        content:
+          application/json:
+            schema:
+              anyOf:
+                - $ref: '#/components/schemas/NewOrder'
```
Comment (tjotala) [inline reference/redox.v1.yaml:25]: This is a bit confusing. We won't be sending a `NewOrder` to Redox, right? We'll be receiving one, and sending one of (`NewResults` | `NewMedia` | `NewNotes`) response back.

But I guess this is just to drive a codegen tool to produce code, it doesn't really matter.

Comment (toddkazakov) [reply reference/redox.v1.yaml:25]: This was the only way to get model generation working. Initially it was no-op endpoint then I saw that Redox exposes a single endpoint for all data models. We could just add what we need here, or even add all models. I didn't try generating a client.

File: reference/redox.v1.yaml (hunk 3 above)
Comment (toddkazakov) [reply reference/redox.v1.yaml:18]: If we use this to generate a client - yes. I'll do this in a follow up task when I get there.

File: reference/redox.v1.yaml (hunk 2 above)
Comment (toddkazakov) [reply reference/redox.v1.yaml:35]: When we get to client generation - yes. I didn't see Bearer auth in stoplight.

Comment (tjotala) [reply reference/redox.v1.yaml:40]: https://swagger.io/docs/specification/authentication/bearer-authentication/

**Pull request merged**